src/generate-embeddings.ts

```diff
@@ -105,22 +109,26 @@ export async function generateEmbeddings(
 
 			// Parse frontmatter and split content into paragraphs
 			const { content, frontmatter } = parseMarkdown(markdown); // Parse the frontmatter
-			const sections = splitIntoParagraphs(content);
+			const sections = smartChunkParagraphs(content, {
+				minParagraphSize: config.minParagraphSize, // Minimum size for a chunk in characters
+				maxParagraphSize: config.maxParagraphSize, // Maximum size for a chunk in characters
+			});
 
-			// Create/update page record. Intentionally clear checksum until we
-			// have successfully generated all page sections.
+			frontmatter["path"] = file.path;
 
+			// Get first chunk context (first sentence or two)
+			// const firstChunkContext = sections[0]?.split(/[.!?][\s\n]+/)?.[0] || '';
 			const { error: upsertPageError, data: document } =
 				await supabaseClient
 					.from("document")
 					.upsert(
 						{
 							checksum: null,
 							path: file.path,
-							meta: JSON.stringify(frontmatter),
+							meta: frontmatter,
```

Comment:
this was an existing bug - it's not required to make a json blob searchable. Not sure if it's going to break anything else if the original author is expecting it to be a json string